.github/workflows/ci.yml

```diff
@@ -19,24 +19,24 @@ jobs:
   test:
     strategy:
       matrix:
-        platform: 
-        - ubuntu-latest
-        go:
-        - 1.18.x
-        - 1.19.x
-        include:
-        - platform: macos-latest
-          go: 1.19.x
-        - platform: windows-latest
-          go: 1.19.x
+        platform: [ ubuntu-latest, windows-latest, macos-latest ]
+        version: [ v1.19.x ]
+        goarch: [ amd64, '386' ]
+        exclude:
+          - platform: macos-latest
+            goarch: '386'
+          - platform: windows-latest
+            goarch: '386'
```

Comment:
This is great, thanks!

We may want to reduce the matrix a bit to a single 32 bit runner in the future, assuming we can prove the effective coverage is the same. As someone who's worked with 32 bit arch's more than me, do you foresee any problems with only validating those arch's on a single platform and Go version?

Reply:
The only reason I didn't also add 32-bit ARM as well is that it's a huge pain to do (without Docker) in GitHub Actions, because the GitHub-hosted runners are all AMD64 currently (ARM64 is coming this fall maybe?).

Another option is to use a self-hosted runner, and I happen to be deep in that rabbit-hole right now (https://github.com/hairyhenderson/gomplate/pull/2126) and it's going about as well as you'd expect (i.e. not at all well).

For this particular class of bug any 32-bit platform will generally do, but there are other classes of bugs that can crop up only on 32-bit ARM CPUs. Somewhat ironically it's extremely rare to run into 32-bit Intel systems except for niche embedded use-cases, but it's still quite common to run into 32-bit ARM given the massive proliferation of older Raspberry Pi-type systems and routers strewn across the IoT landscape.